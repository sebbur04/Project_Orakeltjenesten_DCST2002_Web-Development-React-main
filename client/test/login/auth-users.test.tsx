// Import neccessary modules for testing
import * as React from "react";
import { shallow } from "enzyme";
import Users from "../../src/login/users";
import userService from "../../src/service/user-service";
import permissionService from "../../src/service/permission-service";

jest.mock("../../src/service/user-service");
jest.mock("../../src/service/permission-service");

describe("Users component tests", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("Users renders correctly", (done) => {
    (userService.getAuthenticatedUser as jest.Mock).mockResolvedValue({
      id: 1,
      username: "testUser",
      permid: 1,
    });
    (permissionService.getPermission as jest.Mock).mockResolvedValue({
      users: true,
    });
    (userService.getAllUsers as jest.Mock).mockResolvedValue([
      { id: 2, username: "user1" },
      { id: 3, username: "user2" },
    ]);

    const wrapper = shallow(<Users />);

    setTimeout(() => {
      wrapper.update();
      expect(
        wrapper.containsMatchingElement(<h3>Mangler rettigheter</h3>)
      ).toEqual(false);
      expect(wrapper).toMatchSnapshot();
      done();
    }, 500);
  });

  test("Users correctly handles missing permissions", (done) => {
    (userService.getAuthenticatedUser as jest.Mock).mockResolvedValue({
      id: 1,
      username: "testUser",
      permid: 1,
    });
    (permissionService.getPermission as jest.Mock).mockResolvedValue({
      users: false,
    });

    const wrapper = shallow(<Users />);

    setTimeout(() => {
      wrapper.update();
      expect(
        wrapper.containsMatchingElement(<h3>Mangler rettigheter</h3>)
      ).toEqual(true);
      done();
    }, 500);
  });
});
