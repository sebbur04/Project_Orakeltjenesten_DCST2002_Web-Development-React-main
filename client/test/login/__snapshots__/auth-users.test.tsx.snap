// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component tests Users renders correctly 1`] = `
<div
  className="mainContent"
>
  <div
    className="useredit"
  >
    <Card
      title="Brukere"
    >
      <Row>
        <Column>
          <FormInput
            onChange={[Function]}
            placeholder="SÃ¸k etter bruker"
            type="text"
            value=""
            width="40vw"
          />
        </Column>
      </Row>
      <Row
        key="2"
      >
        <Qe>
          ---
        </Qe>
        <Column
          width={4}
        >
          <a
            className="blue-link"
            href="/#/users/2"
          >
            user1
          </a>
        </Column>
        <Column
          width={3}
        >
          <ButtonLight
            onClick={[Function]}
          >
            Endre rettigheter
          </ButtonLight>
        </Column>
        <Column
          width={2}
        >
          <ButtonDanger
            onClick={[Function]}
          >
            Slett bruker
          </ButtonDanger>
        </Column>
      </Row>
      <Row
        key="3"
      >
        <Qe>
          ---
        </Qe>
        <Column
          width={4}
        >
          <a
            className="blue-link"
            href="/#/users/3"
          >
            user2
          </a>
        </Column>
        <Column
          width={3}
        >
          <ButtonLight
            onClick={[Function]}
          >
            Endre rettigheter
          </ButtonLight>
        </Column>
        <Column
          width={2}
        >
          <ButtonDanger
            onClick={[Function]}
          >
            Slett bruker
          </ButtonDanger>
        </Column>
      </Row>
    </Card>
  </div>
</div>
`;
